### DISCUSSION TITLE
training resize option

### DISCUSSION THREAD

**Discussion by mbenami** [ultralytics/yolov5#7611] [Q&A]
Hi @glenn-jocher  thanks for this awesome repo!

In `letterbox` function there is an option `scaleFill` 
that in case it's True and `auto` False will resize the image to fill the new_shape exactly (no padding) 

I try to change  `scaleFill=True` 
but got poor results ( from mAP0.5 ~90%  got to 80% ) 

1.
in case I'm interested in training without padding, 
is there another code that needs to be changed? 

2.
if I do pad, why there is a need to subtract in `scale_coords` 
for example, if I have 2 images 
A: 500x640
B: 480x640 
both will be resized to 640x640 ( with gray padding) 
hows the model box output coordinate know to take that into account?

3. 
I convert the model to tfjs and the output coordinates are scaled between [0,1] 
while when I check on val.py with the PyTorch weights I get [0, img-size]
is there a reason why on the tfjs the results are scaled? can one convert the model to get the exact coordinate as on PyTorch?

Thanks for your help.

**Comment by glenn-jocher** [comment]
@mbenami you can make any modifications you want obviously, but support for custom work is outside of the context of Ultralytics support. Community users may be able to offer assistance though.
